src/common/Utilities.py

```diff
@@ -463,3 +463,29 @@ def str_to_int_list(str_list):
             int_item = list(range(strt, stop + 1))
         int_list = int_list + int_item
     return int_list
+
+def dot_product_test(operator, num_tests = 20, max_err = 10e-5):
```

Comment:
It sounds like it's testing the dot-product... What about `test_operator_adjoint` or `test_adjoint_operation`? Also, needs some doc on `max_err`

Reply:
Sure, this was a temporary name, as @paskino  said he chatted with some mathematicians and they had some opinions on the correct name of the test, something like "linearity test" but I don't 100% recall.

Reply:
certainly not `linearity_test`. An operator can be linear, while we have a lousy `adjoint` implementation (or even none at all)

Reply:
That is literally what I said in that meeting, but I am waiting for Edo to confirm.